tqdm/_tqdm.py

```diff
@@ -716,12 +797,13 @@ def __iter__(self):
                 # Note: does not call self.update(1) for speed optimisation.
                 n += 1
                 # check the counter first (avoid calls to time())
-                if n - last_print_n >= miniters:
+                if n - last_print_n >= self.miniters:
+                    miniters = self.miniters  # watch monitoring thread changes
```

Comment:
Can we directly modify `self.miniters` or am I missing something? I'm probably missing something.